qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -1157,3 +1157,23 @@ def update_category(self, category, samples_and_values):
                     % (category, value_str, value_types_str, column_type))
 
             raise e
+
+    def check_restrictions(self, restrictions):
+        """Checks if the template fulfills the restrictions
+
+        Parameters
+        ----------
+        restrictions : list of Restriction
+            The restrictions to test if the template fulfills
+
+        Returns
+        -------
+        set of str
+            The missing columns
+        """
+        def _col_iter():
+            for restriction in restrictions:
+                for col in restriction.columns:
+                    yield col
+
```

Comment:
makes sense. in that case though, it would be better to just create the set
using two for loops. eg

cols = set()
for restriction in restrictions:
    for col in restriction.columns:
        cols.add(col)

On Mon, May 4, 2015 at 2:29 PM, josenavas <notifications@github.com> wrote:

> In qiita_db/metadata_template/base_metadata_template.py
> <https://github.com/biocore/qiita/pull/1141#discussion_r29620275>:
>
> > +
> > +        Parameters
> > +        ----------
> > +        restrictions : list of Restriction
> > +            The restrictions to test if the template fulfills
> > +
> > +        Returns
> > +        -------
> > +        set of str
> > +            The missing columns
> > +        """
> > +        def _col_iter():
> > +            for restriction in restrictions:
> > +                for col in restriction.columns:
> > +                    yield col
> > +
>
> I was just avoiding the double list comprehension, as sometimes it has
> been criticized. I think it is not too complicated, so I'll change it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/1141/files#r29620275>.
>